### ISSUE TITLE
Translation from `case_when` to  `fcase()` will throw error when `T` is used instead of `TRUE`.

### ISSUE THREAD

**Issue by NZambranoc** [tidyverse/dtplyr#272]
When calling `T` instead of `TRUE` inside this a case_when using dplyr causes the function to fail.

**This works:**
```r
iris %>% 
lazy_dt() %>%
  mutate(new_col = case_when( Sepal.Length> 4.5 ~ "foo",
                              TRUE ~ "bar"))
```

**This breaks:**
```r

iris %>% 
lazy_dt() %>%
  mutate(new_col = case_when( Sepal.Length> 4.5 ~ "foo",
                              T ~ "bar"))
```
**Error message output to the console:**

```
Error in fcase(Sepal.Length > 4.5, "foo", TRUE, "bar") : 
  Argument #3 has a different length than argument #1. Please make sure all logical conditions have the same length.
````

**Comment by markfairbanks** [comment]
Thanks for catching this. I added a fix in #269 that will address this issue once it's merged.

**Comment by markfairbanks** [comment]
@NZambranoc All set

``` r
library(dplyr, warn.conflicts = FALSE)
library(dtplyr)

iris %>% 
  lazy_dt() %>%
  mutate(new_col = case_when(Sepal.Length > 4.5 ~ "foo",
                             T ~ "bar"))
#> Source: local data table [150 x 6]
#> Call:   copy(`_DT1`)[, `:=`(new_col = fcase(Sepal.Length > 4.5, "foo", 
#>     rep(TRUE, .N), "bar"))]
#> 
#>   Sepal.Length Sepal.Width Petal.Length Petal.Width Species new_col
#>          <dbl>       <dbl>        <dbl>       <dbl> <fct>   <chr>  
#> 1          5.1         3.5          1.4         0.2 setosa  foo    
#> 2          4.9         3            1.4         0.2 setosa  foo    
#> 3          4.7         3.2          1.3         0.2 setosa  foo    
#> 4          4.6         3.1          1.5         0.2 setosa  foo    
#> 5          5           3.6          1.4         0.2 setosa  foo    
#> 6          5.4         3.9          1.7         0.4 setosa  foo    
#> # … with 144 more rows
#> 
#> # Use as.data.table()/as.data.frame()/as_tibble() to access results
```
